workflow/common/validate.go

```diff
@@ -15,21 +15,35 @@ import (
 // wfValidationCtx is the context for validating a workflow spec
 type wfValidationCtx struct {
 	wf *wfv1.Workflow
+	// globalParams keeps track of variables which are available the global
+	// scope and can be referenced from anywhere.
+	globalParams map[string]string
 	// results tracks if validation has already been run on a template
 	results map[string]bool
 }
 
 // ValidateWorkflow accepts a workflow and performs validation against it
 func ValidateWorkflow(wf *wfv1.Workflow) error {
-	err := VerifyUniqueNonEmptyNames(wf.Spec.Templates)
+	ctx := wfValidationCtx{
+		wf:           wf,
+		globalParams: make(map[string]string),
+		results:      make(map[string]bool),
+	}
+
+	err := validateWorkflowFieldNames(wf.Spec.Templates)
+	if err != nil {
+		return errors.Errorf(errors.CodeBadRequest, "spec.templates%s", err.Error())
+	}
+	err = validateArguments(wf.Spec.Arguments)
 	if err != nil {
-		return errors.Errorf(errors.CodeBadRequest, "templates%s", err.Error())
+		return errors.Errorf(errors.CodeBadRequest, "spec.arguments%s", err.Error())
 	}
 
-	ctx := wfValidationCtx{
-		wf:      wf,
-		results: make(map[string]bool),
+	ctx.globalParams["workflow.name"] = "placeholder-workflow-name"
```

Comment:
Can "placeholder-workflow-name" be a constant declared elsewhere?

Reply:
OK